core/plugins/org.polarsys.capella.core.model.helpers/src/org/polarsys/capella/core/model/helpers/move/MoveHelper.java

```diff
@@ -335,6 +350,183 @@ protected boolean checkCompatibility(List<EObject> modelElements, EList<EReferen
     }
     return areCompatible;
   }
+  
+  /**
+   * Populate 'states' parameter with all states of a region
+   * 
+   * @param region
+   *          Region
+   * @param states
+   *          List with all encountered states - at the beginning is empty - it will be updated by appending all the
+   *          states of the region
+   */
+  private void populateListWithAllRegionStates(Region region, EList<AbstractState> states) {
+    EList<AbstractState> ownedRegionStates = region.getOwnedStates();
+    if (ownedRegionStates == null || ownedRegionStates.isEmpty())
+      return;
+    states.addAll(ownedRegionStates);
+    for (AbstractState state : ownedRegionStates) {
+      if (state instanceof State) {
+        EList<Region> ownedRegions = ((State) state).getOwnedRegions();
+        if (!(ownedRegions.isEmpty()) && (ownedRegions != null)) {
+          for (Region ownedRegion : ownedRegions) {
+            populateListWithAllRegionStates(ownedRegion, states);
+          }
+        }
+      }
+    }
+  }
+
+  /**
+   * Get all states of a region
+   * 
+   * @param region
+   *          Region
+   * @return all states of a region
+   */
+  private EList<AbstractState> getAllRegionStates(Region region) {
```

Comment:
org.polarsys.capella.core.model.helpers.move.MoveHelper.getAllModeState(Region) does not work?

Reply:
you are right, i deleted it